dev/com.ibm.ws.concurrent.mp.1.0/src/com/ibm/ws/concurrent/mp/spi/MPContextPropagation.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.mp.spi;
+
+import org.eclipse.microprofile.context.ManagedExecutor;
+import org.eclipse.microprofile.context.ThreadContext;
+
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.concurrent.mp.ManagedExecutorImpl;
+import com.ibm.ws.concurrent.mp.ThreadContextImpl;
+import com.ibm.ws.threading.PolicyExecutor;
+import com.ibm.wsspi.kernel.service.utils.AtomicServiceReference;
+
+/**
+ * Provides static methods to enable the ManagedExecutorBuilderImpl and ThreadContextBuilderImpl,
+ * (both of which are from another bundle) to create ManagedExecutor and ThreadContext instances.
+ */
+@Trivial
+public class MPContextPropagation {
+    /**
+     * Creates a new MicroProfile ManagedExecutor instance.
+     *
+     * @param managedExecutorName    unique name for the new ManagedExecutor instance.
+     * @param threadContextName      unique name for a new ThreadContext instance to be used by the new ManagedExecutor instance.
+     * @param hash                   hash code for the new instance.
+     * @param policyExecutor         Executor that runs tasks under the concurrency policy for this managed executor.
+     * @param config                 represents thread context propagation configuration.
+     * @param tranContextProviderRef Reference to the transaction context provider.
+     * @return the new instance.
+     */
+    public static ManagedExecutor createManagedExecutor(String managedExecutorName, String threadContextName, int hash,
+                                                        PolicyExecutor policyExecutor, ThreadContextConfig config,
+                                                        @SuppressWarnings("deprecation") AtomicServiceReference<com.ibm.wsspi.threadcontext.ThreadContextProvider> tranContextProviderRef) {
+        ThreadContextImpl mpThreadContext = new ThreadContextImpl(threadContextName, hash, config);
```

Comment:
Should this just call down to the createThreadContext so there is only ever one place to change if you need to update the creation of a ThreadContext?

Reply:
Constructing as ThreadContextImpl is more convenient here because createThreadContext returns the value as a ThreadContext which would require a cast to WSContextService.  But if using ThreadContextImpl, it is already both.